src/app/mat-select-search/mat-select-search.component.ts

```diff
@@ -499,40 +493,14 @@ export class MatSelectSearchComponent implements OnInit, OnDestroy, ControlValue
       const ariaActiveDescendantId = this.matSelect._getAriaActiveDescendant();
       const index = this._options.toArray().findIndex(item => item.id === ariaActiveDescendantId);
       if (index !== -1) {
-        const activeDescendant = this._options.toArray()[index];
-        this.liveAnnouncer.announce(
-          activeDescendant.viewValue + ' '
-          + this.getAriaIndex(index)
-          + this.indexAndLengthScreenReaderText
-          + this.getAriaLength()
-        );
+        this.activeDescendant?.removeAttribute('aria-selected');
+        this.activeDescendant = this._options.toArray()[index]._getHostElement();
```

Comment:
Please use a WeakRef here:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakRef

Reply:
I refactored the code to use a WeakRef, but apparently the CI build step `build-lib` is failing. To support the WeakRef, I had to upgrade the `lib` from es2017 to es2021. 

According to [Can I use](https://caniuse.com/mdn-javascript_builtins_weakref) the feature is not included in every browser version. I tried updating the `.browserslistrc` according to the table form Can I use, but the build would still fail. 

Personally, I haven't used the `ng-packagr` library before. Maybe you could help to get the build step working?

Reply:
Since we only keep one reference per component this would propably not result in a lot of memory drain. So I think rolling back with the changes is okay in our situation.

I refactored the code again to use a solid reference like before. With that the build is now working and I would say the pull request is ready to be merged.